File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnPlaylistAttributes.java

package com.gracenote.gnsdk;

/** 
* <b>Experimental</b>: {@link GnPlaylistAttributes} 
*/ 
 
public class GnPlaylistAttributes extends GnDataObject {
  private long swigCPtr;

  protected GnPlaylistAttributes(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnPlaylistAttributes_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnPlaylistAttributes obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnPlaylistAttributes(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnPlaylistAttributes(GnPlaylistAttributes other) {
    this(gnsdk_javaJNI.new_GnPlaylistAttributes(GnPlaylistAttributes.getCPtr(other), other), true);
  }

/** 
*  Retrieves the AlbumName value as a string for this attribute . 
*  @return album name string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String albumName() {
    return gnsdk_javaJNI.GnPlaylistAttributes_albumName(swigCPtr, this);
  }

/** 
*  Retrieves the ArtistName value as a string for this attribute . 
*  @return artist name string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String artistName() {
    return gnsdk_javaJNI.GnPlaylistAttributes_artistName(swigCPtr, this);
  }

/** 
*  Retrieves the ArtistType value as a string for this attribute . 
*  @return artist type string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String artistType() {
    return gnsdk_javaJNI.GnPlaylistAttributes_artistType(swigCPtr, this);
  }

/** 
*  Retrieves the Era value as a string for this attribute . 
*  @return era string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String era() {
    return gnsdk_javaJNI.GnPlaylistAttributes_era(swigCPtr, this);
  }

/** 
*  Retrieves the Genre value as a string for this attribute . 
*  @return genre string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String genre() {
    return gnsdk_javaJNI.GnPlaylistAttributes_genre(swigCPtr, this);
  }

/** 
*  Retrieves the Origin value as a string for this attribute . 
*  @return origin string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String origin() {
    return gnsdk_javaJNI.GnPlaylistAttributes_origin(swigCPtr, this);
  }

/** 
*  Retrieves the Mood value as a string for this attribute . 
*  @return mood string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String mood() {
    return gnsdk_javaJNI.GnPlaylistAttributes_mood(swigCPtr, this);
  }

/** 
*  Retrieves the Tempo value as a string for this attribute . 
*  @return tempo string value if it exists else an empty string. 
* <p> 
**/ 
 
  public String tempo() {
    return gnsdk_javaJNI.GnPlaylistAttributes_tempo(swigCPtr, this);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnPlaylistAttributes.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* <b>Experimental</b>: GnPlaylistAttributes
*/
public class GnPlaylistAttributes : GnDataObject {
  private HandleRef swigCPtr;

  internal GnPlaylistAttributes(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnPlaylistAttributes obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnPlaylistAttributes() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnPlaylistAttributes(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnPlaylistAttributes(GnPlaylistAttributes other) : this(gnsdk_csharp_marshalPINVOKE.new_GnPlaylistAttributes(GnPlaylistAttributes.getCPtr(other)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Retrieves the AlbumName value as a string for this attribute .
*  @return album name string value if it exists else an empty string.
*
**/
  public string AlbumName {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_AlbumName_get(swigCPtr) );
	} 

  }

/**
*  Retrieves the ArtistName value as a string for this attribute .
*  @return artist name string value if it exists else an empty string.
*
**/
  public string ArtistName {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_ArtistName_get(swigCPtr) );
	} 

  }

/**
*  Retrieves the ArtistType value as a string for this attribute .
*  @return artist type string value if it exists else an empty string.
*
**/
  public string ArtistType {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_ArtistType_get(swigCPtr) );
	} 

  }

/**
*  Retrieves the Era value as a string for this attribute .
*  @return era string value if it exists else an empty string.
*
**/
  public string Era {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_Era_get(swigCPtr) );
	} 

  }

/**
*  Retrieves the Genre value as a string for this attribute .
*  @return genre string value if it exists else an empty string.
*
**/
  public string Genre {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_Genre_get(swigCPtr) );
	} 

  }

/**
*  Retrieves the Origin value as a string for this attribute .
*  @return origin string value if it exists else an empty string.
*
**/
  public string Origin {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_Origin_get(swigCPtr) );
	} 

  }

/**
*  Retrieves the Mood value as a string for this attribute .
*  @return mood string value if it exists else an empty string.
*
**/
  public string Mood {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_Mood_get(swigCPtr) );
	} 

  }

/**
*  Retrieves the Tempo value as a string for this attribute .
*  @return tempo string value if it exists else an empty string.
*
**/
  public string Tempo {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylistAttributes_Tempo_get(swigCPtr) );
	} 

  }

}

}
